src/main/groovy/com/google/gcloud/exec/command/CommandBuilder.groovy

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright 2015 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.gcloud.exec.command
+
+/**
+ * build a gcloud command
+ */
+
+class CommandBuilder {
+    List<String> command = []
+    List<String> options = []
+    List<String> arguments = []
+
+    public CommandBuilder(String... parts) {
+        Collections.addAll(command, parts);
+    }
+
+    /**
+     * Returns a command String[] of the form command + options + arguments
+     */
+    public String[] buildCommand() {
+        if (command.size() == 0) {
+            throw new RuntimeException("No Command Specified");
+        }
+        List<String> commandLine = []
+        commandLine.addAll(command)
+        commandLine.addAll(options)
+        commandLine.addAll(arguments)
+
+        return commandLine.toArray(new String[commandLine.size()])
+    }
+
+    public CommandBuilder addCommand(String... parts) {
+        command.addAll(parts)
+        return this;
+    }
+
+    /**
+     * Convenience method to add non-null values as options to a gcloud command
+     * in the format "--optionName=value" or "--optionName", does NOT escape special characters
+     */
+    public CommandBuilder addOption(String optionName, String value) {
+        if (optionName == null || optionName.trim().isEmpty()) {
+            throw new IllegalArgumentException("Option is null or empty")
+        }
+        if (value == null || value.trim().isEmpty() || value.trim().equals("true")) {
+            if (optionName.startsWith("--")) {
```

Comment:
Probably not worth the effort, but what do you think about rejecting single hypen'd option names so the user of this library can't accidentally create an option like ---c=3